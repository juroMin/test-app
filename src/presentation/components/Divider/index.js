import React from 'react';
import {View, StyleSheet} from 'react-native';
import {divider} from 'colors';

export default ({style}) => {
  return <View style={[styles.container, style]} />;
};

const styles = StyleSheet.create({
  container: {
    height: 2,
    backgroundColor: divider,
  },
});
