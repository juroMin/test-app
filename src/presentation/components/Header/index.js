import React from 'react';
import {View, Image} from 'react-native';
import {Divider} from 'components';
import {logo} from 'assets';
import styles from './styles';

export default () => (
  <>
    <View style={styles.container}>
      <Image source={logo} resizeMode="contain" style={styles.logo} />
    </View>
    <Divider />
  </>
);
