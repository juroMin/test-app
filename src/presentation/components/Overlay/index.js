import React from 'react';
import {View, StyleSheet} from 'react-native';

export default ({style, ...props}) => (
  <View
    {...props}
    pointerEvents={props.pointerEvents || 'box-none'}
    style={[styles.basic, props.centered ? styles.centered : null, style]}
  />
);

const styles = StyleSheet.create({
  basic: {
    position: 'absolute',
    left: 0,
    right: 0,
    bottom: 0,
    top: 0,
  },
  centered: {
    alignItems: 'center',
    justifyContent: 'center',
  },
});
