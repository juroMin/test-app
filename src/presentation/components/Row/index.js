import React from 'react';
import {View, StyleSheet} from 'react-native';

export default ({style, ...props}) => (
  <View
    {...props}
    style={[styles.basic, props.centered && styles.centered, style]}
  />
);

const styles = StyleSheet.create({
  basic: {
    flexDirection: 'row',
  },
  centered: {
    justifyContent: 'center',
  },
});
