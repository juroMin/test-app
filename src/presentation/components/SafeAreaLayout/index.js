import React from 'react';
import {useSafeArea} from 'react-native-safe-area-context';
import {View} from 'react-native';
import {SaveAreaInset} from 'enums';
import {appBackground} from 'colors';

export const SafeAreaLayout = (props) => {
  const safeAreaInsets = useSafeArea();
  const {insets, style, ...layoutProps} = props;

  const toStyleProp = (inset) => {
    switch (inset) {
      case SaveAreaInset.BOTTOM:
        return {paddingBottom: safeAreaInsets.bottom};
      case SaveAreaInset.TOP:
        return {paddingTop: safeAreaInsets.top};
      default:
        return {
          paddingTop: safeAreaInsets.top,
          paddingBottom: safeAreaInsets.bottom,
        };
    }
  };

  const createInsets = () => {
    if (!insets) {
      return toStyleProp();
    } else {
      return React.Children.map(insets, toStyleProp);
    }
  };

  return (
    <View
      {...layoutProps}
      style={[{backgroundColor: appBackground}, style, createInsets()]}
    />
  );
};
