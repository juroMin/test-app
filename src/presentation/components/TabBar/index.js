import React from 'react';
import {View, TouchableWithoutFeedback, Animated, Easing} from 'react-native';
import {SafeAreaLayout} from 'components';
import {SaveAreaInset} from 'enums';
import {primary, accented} from 'colors';
import styles from './styles';

export default ({state, descriptors, navigation}) => {
  return (
    <SafeAreaLayout insets={SaveAreaInset.BOTTOM} style={styles.mainContainer}>
      {state.routes.map((route, index) => {
        const {options} = descriptors[route.key];
        const isFocused = state.index === index;
        const TabIcon = options.tabBarIcon;
        let scaleValue = new Animated.Value(0);

        function scale() {
          scaleValue.setValue(0);
          Animated.timing(scaleValue, {
            toValue: 1,
            duration: 300,
            easing: Easing.easeOutBack,
            useNativeDriver: false,
          }).start();
        }

        const onPress = () => {
          scale(scaleValue);
          const event = navigation.emit({
            type: 'tabPress',
            target: route.key,
          });

          if (!isFocused && !event.defaultPrevented) {
            navigation.navigate(route.name);
          }
        };

        const onLongPress = () => {
          scale(scaleValue);
          navigation.emit({type: 'tabLongPress', target: route.key});
        };

        const buttonScale = scaleValue.interpolate({
          inputRange: [0, 0.5, 1],
          outputRange: [1, 1.5, 1],
        });

        return (
          <View key={index} style={styles.tabContainer}>
            <TouchableWithoutFeedback
              accessibilityRole="button"
              style={styles.clickableArea}
              testID={options.tabBarTestID}
              onPress={onPress}
              onLongPress={onLongPress}>
              <Animated.View
                style={[styles.tab, {transform: [{scale: buttonScale}]}]}>
                <TabIcon
                  size={styles.iconSize}
                  marginVertical={2}
                  tintColor={isFocused ? accented : primary}
                />
              </Animated.View>
            </TouchableWithoutFeedback>
          </View>
        );
      })}
    </SafeAreaLayout>
  );
};
