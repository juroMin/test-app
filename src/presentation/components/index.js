export {default as Row} from './Row';
export {default as Text} from './Text';
export {default as Divider} from './Divider';
export {default as Header} from './Header';
export {SafeAreaLayout} from './SafeAreaLayout';
export {default as TabBar} from './TabBar';
export {default as Overlay} from './Overlay';
