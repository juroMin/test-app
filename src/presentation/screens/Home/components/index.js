export {default as DetailsSection} from './DetailsSection';
