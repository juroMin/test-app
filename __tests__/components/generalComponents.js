import React from 'react';
import renderer from 'react-test-renderer';
import {Divider, Header, Text, Row, SafeAreaLayout, Overlay} from 'components';

test('Divider renders correctly', () => {
  const tree = renderer.create(<Divider />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Header renders correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Text renders correctly', () => {
  const tree = renderer.create(<Text />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Row renders correctly', () => {
  const tree = renderer.create(<Row />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Safe Area Layout renders correctly', () => {
  const tree = renderer.create(<SafeAreaLayout />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Overlay renders correctly', () => {
  const tree = renderer.create(<Overlay />).toJSON();
  expect(tree).toMatchSnapshot();
});
