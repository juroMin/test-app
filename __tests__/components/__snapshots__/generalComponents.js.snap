// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f3f3f3",
        "height": 2,
      },
      undefined,
    ]
  }
/>
`;

exports[`Header renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "marginVertical": 10,
      }
    }
  >
    <Image
      resizeMode="contain"
      style={
        Object {
          "height": 50,
          "width": 150,
        }
      }
    />
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#f3f3f3",
          "height": 2,
        },
        undefined,
      ]
    }
  />,
]
`;

exports[`Overlay renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      null,
      undefined,
    ]
  }
/>
`;

exports[`Row renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`Safe Area Layout renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
      },
      undefined,
      Object {
        "paddingBottom": undefined,
        "paddingTop": undefined,
      },
    ]
  }
/>
`;

exports[`Text renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "black",
        "fontSize": 14,
        "textAlign": "justify",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
/>
`;
